buildpack.md

```diff
@@ -162,12 +166,38 @@ Executable: `/bin/build <layers[EIC]> <platform[AR]> <plan[ER]>`, Working Dir: `
 
 Using the [Layer Content Metadata](#layer-content-metadata-toml) provided by a buildpack in a `<layers>/<layer>.toml` file, the lifecycle MUST determine:
 
-- Whether the layer directory in `<layers>/<layer>/` should be available to the app (via the `launch` boolean).
-- Whether the layer directory in `<layers>/<layer>/` should be available to subsequent buildpacks (via the `build` boolean).
-- Whether and how the layer directory in `<layers>/<layer>/` should be persisted to subsequent builds of the same OCI image (via the `cache` boolean).
+- Whether the layer directory `<layers>/<layer>/` should be available to the app (via the `launch` boolean).
+- Whether the layer directory `<layers>/<layer>/` should be available to subsequent buildpacks (via the `build` boolean).
+- Whether the layer directory `<layers>/<layer>/` should be persisted and made available to subsequent builds of the same OCI image (via the `cache` boolean).
 
 All combinations of `launch`, `build`, and `cache` booleans are valid. When a layer declares more than one type (e.g. `launch = true` and `cache = true`), the requirements of each type apply.
 
+The following table illustrates the behavior depending on the value of each flag.
+Note that the lifecycle only restores layers from the cache due to performance reasons.
+
+`build`   | `cache`  | `launch` | Metadata Restored        | Layer Restored      
+----------|----------|----------|--------------------------|---------------------
+T         | T        | T        | Yes - from the app image | Yes* - from the cache
+T         | T        | F        | Yes - from the cache     | Yes - from the cache
+T         | F        | T        | No                       | No
+T         | F        | F        | No                       | No
+F         | T        | T        | Yes - from the app image | Yes* - from the cache
+F         | T        | F        | Yes - from the cache     | Yes - from the cache
+F         | F        | T        | Yes - from the app image | No
+F         | F        | F        | No                       | No
+
+\* The metadata and layer are restored only if the layer SHA recorded in the previous image matches the layer SHA recorded in the cache.
+
+Examples:
```

Comment:
I like the examples here! We historically haven't done examples in the spec. @sclevine @ekcasey is this something we want to keep here or move somewhere else (in that reference doc that we keep talking about).

Reply:
Hi @hone, I was also wondering on whether this is the right place for it (although it felt right :) ) so we had [this](https://buildpacks.slack.com/archives/CCP60GJLS/p1616444461016800) conversation over slack and @sclevine and @ekcasey agreed that it's a good idea to add the examples here.

Reply:
hi @yaelharel. I'm very 👍 for examples fwiw, I didn't see that slack convo. It does make the rest of the spec look really lame in comparison. ;) I'll mark this as resolved.